File: query/sql/plsql/JavaScript/PlSqlParserBase.js
import { Parser } from "antlr4";

export default class PlSqlParserBase extends Parser {
  _isVersion10;
  _isVersion12;

  constructor(input) {
    super(input);
    this._isVersion10 = false;
    this._isVersion12 = true;
  }

  isVersion10() {
    return this._isVersion10;
  }

  isVersion12() {
    return this._isVersion12;
  }

  setVersion10(value) {
    this._isVersion10 = value;
  }

  setVersion12(value) {
    this._isVersion12 = value;
  }
}

File: query/sql/plsql/Dart/PlSqlParserBase.dart
import 'package:antlr4/antlr4.dart';
import 'dart:io';
import 'dart:convert';

abstract class PlSqlParserBase extends Parser
{
    PlSqlParserBase self;
    bool _isVersion12 = true;
    bool _isVersion10 = true;

    PlSqlParserBase(TokenStream input) : super(input)
    {
        self = this;
    }

    bool isVersion12() {
        return _isVersion12;
    }

    void setVersion12(bool value) {
        _isVersion12 = value;
    }

    bool isVersion10() {
        return _isVersion10;
    }

    void setVersion10(bool value) {
        _isVersion10 = value;
    }
}
